cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkContextConfig.java

```diff
@@ -91,47 +96,55 @@ public String getExecutionMode() {
   }
 
   /**
-   * Returns the {@link SparkSpecification} stored in the configuration.
+   * @return the {@link SparkSpecification} stored in the configuration.
    */
   public SparkSpecification getSpecification() {
     return GSON.fromJson(hConf.get(HCONF_ATTR_PROGRAM_SPEC), SparkSpecification.class);
   }
 
   /**
-   * Returns the {@link Id.Program} stored in the configuration.
+   * @return the {@link Id.Program} stored in the configuration.
    */
   public Id.Program getProgramId() {
     return GSON.fromJson(hConf.get(HCONF_ATTR_PROGRAM_ID), Id.Program.class);
   }
 
   /**
-   * Returns the {@link RunId} stored in the configuration.
+   * @return the {@link RunId} stored in the configuration.
    */
   public RunId getRunId() {
     return RunIds.fromString(hConf.get(HCONF_ATTR_RUN_ID));
   }
 
   /**
-   * Returns the runtime arguments stored in the configuration.
+   * @return the runtime arguments stored in the configuration.
    */
   public Map<String, String> getArguments() {
     return GSON.fromJson(hConf.get(HCONF_ATTR_ARGS), ARGS_TYPE);
   }
 
   /**
-   * Returns the logical start time stored in the configuration.
+   * @return the logical start time stored in the configuration.
    */
   public long getLogicalStartTime() {
     return hConf.getLong(HCONF_ATTR_LOGICAL_START_TIME, System.currentTimeMillis());
   }
 
   /**
-   * Returns the transaction stored in the configuration.
+   * @return the {@link Transaction} stored in the configuration.
    */
   public Transaction getTransaction() {
     return GSON.fromJson(hConf.get(HCONF_ATTR_NEW_TX), Transaction.class);
   }
 
+  /**
+   * @return the {@link WorkflowToken} stored in the configuration.
+   */
+  @Nullable
+  public WorkflowToken getWorkflowToken() {
+    return GSON.fromJson(hConf.get(HCONF_ATTR_WORKFLOW_TOKEN), BasicWorkflowToken.class);
```

Comment:
No, tried a small program:
```
public static void main(String [] args) {
    Gson gson = new Gson();
    gson.toJson(null);
    System.out.println("hey");
    String json = null;
    gson.fromJson(json, BasicWorkflowToken.class);
    System.out.println("dingdong");
  }
```

No exceptions, ``GSON`` seems to handle it fine.